src/plugins/validation/2and3/semantic-validators/operation-ids.js

```diff
@@ -153,15 +155,18 @@ module.exports.validate = function({ resolvedSpec }) {
           regex.test(op['pathKey'])
         );
 
-        if (checkPassed === false) {
-          messages.addMessage(
-            op.path + '.operationId',
-            `operationIds should follow consistent naming convention. operationId verb should be ${verbs}`.replace(
-              ',',
-              ' or '
-            ),
-            'warning'
-          );
+        const checkStatus = config.operationId_naming_convention || 'off';
+        if (checkStatus.match('error|warning')) {
+          if (checkPassed === false) {
+            messages.addMessage(
+              op.path + '.operationId',
+              `operationIds should follow consistent naming convention. operationId verb should be ${verbs}`.replace(
+                ',',
+                ' or '
+              ),
+              checkStatus
+            );
+          }
```

Comment:
A couple of comments - the config handler will automatically fill in all rule values with "error", "warning", or "off", so there's no need for the `|| 'off'`.

Also, much of this logic can be simplified by just passing the key directly to the message carrier - it has built-in checking for whether something is an error, warning, or off:

```suggestion
        if (checkPassed === false) {
          messages.addMessage(
            op.path + '.operationId',
            `operationIds should follow consistent naming convention. operationId verb should be ${verbs}`.replace(
              ',',
              ' or '
            ),
            config.operation_id_naming_convention
          );
        }
```